src/main/java/appeng/client/gui/implementations/GuiFluidTerminal.java

```diff
@@ -0,0 +1,362 @@
+import appeng.client.gui.widgets.GuiScrollbar;
+import appeng.client.gui.widgets.ISortSource;
+import appeng.client.gui.widgets.MEGuiTextField;
+import appeng.client.me.FluidRepo;
+import appeng.client.me.SlotFluidME;
+import appeng.client.me.InternalFluidSlotME;
+import appeng.container.implementations.ContainerFluidTerminal;
+import appeng.container.slot.ISlotFluid;
+import appeng.core.AELog;
+import appeng.core.localization.GuiText;
+import appeng.core.sync.network.NetworkHandler;
+import appeng.core.sync.packets.PacketInventoryAction;
+import appeng.core.sync.packets.PacketValueConfig;
+import appeng.helpers.InventoryAction;
+import appeng.parts.reporting.PartFluidTerminal;
+import appeng.util.IConfigManagerHost;
+import appeng.util.Platform;
+import appeng.util.item.AEFluidStack;
+
+
+/**
+ * @author BrockWS
+ * @version rv6 - 12/05/2018
+ * @since rv6 12/05/2018
+ */
+public class GuiFluidTerminal extends AEBaseMEGui implements ISortSource, IConfigManagerHost
+{
+	private final List<SlotFluidME> meFluidSlots = new LinkedList<>();
+	private final FluidRepo repo;
+	private final IConfigManager configSrc;
+	private final ContainerFluidTerminal container;
+	private final int offsetX = 9;
+	private int rows = 6;
+	private int maxRows = Integer.MAX_VALUE;
+	private int perRow = 9;
+
+	protected PartFluidTerminal terminal;
+
+	private MEGuiTextField searchField;
+	private GuiImgButton sortByBox;
+	private GuiImgButton sortDirBox;
+
+	public GuiFluidTerminal( InventoryPlayer inventoryPlayer, PartFluidTerminal terminal )
+	{
+		super( new ContainerFluidTerminal( inventoryPlayer, terminal ) );
+		this.terminal = terminal;
+		this.xSize = 185;
+		this.ySize = 222;
+		final GuiScrollbar scrollbar = new GuiScrollbar();
+		this.setScrollBar( scrollbar );
+		this.repo = new FluidRepo( scrollbar, this );
+		this.configSrc = ( (IConfigurableObject) this.inventorySlots ).getConfigManager();
+		( this.container = (ContainerFluidTerminal) this.inventorySlots ).setGui( this );
+	}
+
+	@Override
+	public void initGui()
+	{
+		this.mc.player.openContainer = this.inventorySlots;
+		this.guiLeft = ( this.width - this.xSize ) / 2;
+		this.guiTop = ( this.height - this.ySize ) / 2;
+
+		this.searchField = new MEGuiTextField( this.fontRenderer, this.guiLeft + Math.max( 80, this.offsetX ), this.guiTop + 4, 90, 12 );
+		this.searchField.setEnableBackgroundDrawing( false );
+		this.searchField.setMaxStringLength( 25 );
+		this.searchField.setTextColor( 0xFFFFFF );
+		this.searchField.setSelectionColor( 0xFF99FF99 );
+		this.searchField.setVisible( true );
+
+		int offset = this.guiTop;
+
+		this.buttonList.add( this.sortByBox = new GuiImgButton( this.guiLeft - 18, offset, Settings.SORT_BY, this.configSrc.getSetting( Settings.SORT_BY ) ) );
+		offset += 20;
+
+		this.buttonList.add( this.sortDirBox = new GuiImgButton( this.guiLeft - 18, offset, Settings.SORT_DIRECTION, this.configSrc.getSetting( Settings.SORT_DIRECTION ) ) );

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)